pkg/nodelabels/builder.go

```diff
@@ -22,13 +22,15 @@ import (
 )
 
 const (
-	RoleLabelName15        = "kubernetes.io/role"
-	RoleLabelName16        = "kubernetes.io/role"
-	RoleMasterLabelValue15 = "master"
-	RoleNodeLabelValue15   = "node"
+	RoleLabelName15           = "kubernetes.io/role"
+	RoleLabelName16           = "kubernetes.io/role"
+	RoleMasterLabelValue15    = "master"
+	RoleAPIServerLabelValue15 = "api-server"
+	RoleNodeLabelValue15      = "node"
 
-	RoleLabelMaster16 = "node-role.kubernetes.io/master"
-	RoleLabelNode16   = "node-role.kubernetes.io/node"
+	RoleLabelMaster16    = "node-role.kubernetes.io/master"
+	RoleLabelAPIServer16 = "node-role.kubernetes.io/api-server"
```

Comment:
If we're in doubt, we could just label with "control-plane" and add labels in future so we can easily distinguish etcd and non-etcd.  It's much easier to add later than it is to take away something once we've shipped it.

Reply:
We are inventing these here. I couldn't find anyone else with a similar setup, and also discussed a bit with @rifelpet what they had been using.

I think having the role visible in `kubectl get nodes` is useful, but it is not a must.

More importantly is that APIserver nodes may not have all the IAM privileges that the "full" CP nodes have. So it could break things that use this label as node selector.

Reply:
I recall from office hours this week that we were leaning towards using either `apiserver` or `kube-apiserver` for this label.

Reply:
uh, we went a bit back and forth on that. I think `api-server` was preferred as it matched `control-plane`. And specifically we did not want `kube-apiserver`?

Reply:
My thought is that it should either be `kube-apiserver` or `apiserver`, because we should match `kube-apiserver`.  But then we also have `kube-controller-manager` so maybe there is no pattern and we should just go with what we think :-)

Reply:
We are behind a feature flag now, so we can choose one and see how people react.